.python_package/noxfile.py

```diff
@@ -0,0 +1,123 @@
+# GridTools
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import configparser
+import pathlib
+import shutil
+
+import nox
+
+nox.options.sessions = ["test_src", "test_wheel"]
+
+
+def prepare(session: nox.Session):
+    session.install("cmake>=2.18.1")
+    session.install("ninja")
+    build_path = session.cache_dir.joinpath("build").absolute()
+    build_path.mkdir(exist_ok=True)
+    install_path = pathlib.Path(".").absolute() / "src" / "gridtools" / "data"
+    source_path = pathlib.Path("..").absolute()
+    with session.chdir(build_path):
+        session.run(
+            "cmake",
+            "-DBUILD_TESTING=OFF",
+            "-DGT_INSTALL_EXAMPLES:BOOL=OFF",
+            f"-DCMAKE_INSTALL_PREFIX={install_path}",
+            "-GNinja",
+            str(source_path),
+        )
+        session.run("cmake", "--install", ".")
+        session.log("installed gridttols sources")
+    version_path = source_path / "version.txt"
+    setup_path = pathlib.Path(".") / "setup.cfg"
+    config = configparser.ConfigParser()
+    config.read(str(setup_path))
+    config["metadata"]["version"] = version_path.read_text()
+    with setup_path.open("w") as setup_fp:
+        config.write(setup_fp)
+    session.log("updated version metadata")
+    shutil.copy(source_path / "LICENSE", ".")
+    session.log("copied license file")
+
+
+def get_wheel(session: nox.Session) -> pathlib.Path:
+    return list(session.cache_dir.joinpath("dist").glob("gridtools_cpp-*.whl"))[0]
+
+
+@nox.session
+def build_wheel(session: nox.Session):
+    prepare(session)
+    dist_path = session.cache_dir.joinpath("dist").absolute()
+    workdir = pathlib.Path(".").absolute()
+    session.install("build[virtualenv]")
```

Comment:
Maybe it would be good to add all packages used or installed here (`build`, `cmake`, ...) to the list of `build_system.requires` in pyproject.toml to make the dependencies explicit.

Reply:
We could add them for documentation purposes, even they are not build requiements anymore. They are pre-build requirements. In the PEP-517 conformant isolated build environment, they don't have to be present.

A compromise might be to add them commented-out, I guess?